platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingSessionDataStore.java

```diff
@@ -0,0 +1,143 @@
+ * attributes.
+ *
+ * <p>There is one <code>SessionDataStore</code> instance for each HTTP context on the Jetty server.
+ * In the current way we use pax-web with our jax-rs servers, session attributes are not shared
+ * across contexts. Our SSO solution necessitates a way to share those session attributes since the
+ * security token is stored on the session as an attribute.
+ */
+public class AttributeSharingSessionDataStore extends AbstractSessionDataStore {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AttributeSharingSessionDataStore.class);
+
+  private final Map<String, SessionData> sessionDataMap = new ConcurrentHashMap<>();
+  private AttributeSharingHashSessionIdManager attributeSharingHashSessionIdManager;
+
+  /**
+   * Called by the {@link AttributeSharingHashSessionIdManager} whenever it updates one of its
+   * session's attributes.
+   *
+   * @param id the session id
+   * @param sessionAttributes the session's attributes
+   */
+  public void updateSessionAttributes(String id, Map<String, Object> sessionAttributes) {
+    SessionData sessionData = sessionDataMap.get(id);
+    if (sessionData != null && !sessionData.getAllAttributes().equals(sessionAttributes)) {
+      LOGGER.debug(
+          "Storing new attributes for session {} at context {}",
+          id,
+          _context.getCanonicalContextPath());
+      synchronized (this) {
+        sessionData.clearAllAttributes();
+        sessionData.putAllAttributes(sessionAttributes);
+      }
+    }
+  }
+
+  @Override
+  public void initialize(SessionContext context) throws Exception {
+    super.initialize(context);
+    this.attributeSharingHashSessionIdManager =
+        (AttributeSharingHashSessionIdManager) context.getSessionHandler().getSessionIdManager();
+    attributeSharingHashSessionIdManager.addSessionDataStore(this);
+  }
+
+  @Override
+  public void doStore(String id, SessionData data, long lastSaveTime) {
+    attributeSharingHashSessionIdManager.provideNewSessionAttributes(
+        this, id, data.getAllAttributes());
+
+    synchronized (this) {
+      sessionDataMap.put(id, data);
+    }
+  }
+
+  @Override
+  public SessionData newSessionData(
+      String id, long created, long accessed, long lastAccessed, long maxInactiveMs) {
+    SessionData sessionData =
+        new SessionData(
+            id,
+            _context.getCanonicalContextPath(),
+            _context.getVhost(),
+            created,
+            accessed,
+            lastAccessed,
+            maxInactiveMs);
+
+    Map<String, Object> sessionAttributes =
+        attributeSharingHashSessionIdManager.getLatestSessionAttributes(id);
+    if (sessionAttributes != null) sessionData.putAllAttributes(sessionAttributes);
+
+    return sessionData;
+  }
+
+  @Override
+  public Set<String> doGetExpired(Set<String> candidates) {
+    final long now = System.currentTimeMillis();
+
+    return candidates
+        .stream()
+        .filter(
+            c -> {
```

Comment:
✏️ Easier to read if you create a private method named something like `isSessionNotExpired` and reference that as your filter.